### PR TITLE
Sign in - Target MVD

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello board reference:

* [As a user I should be able to login so I can use my Target MVD account](https://trello.com/c/QiDQlXCX)

---

#### Description:

* Add request spec test to validate *sign in* functionality

### CODE REVIEW

File: spec/request/api/v1/auth/sign_in_spec.rb [hunk 1]
```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+require 'faker'
+
+describe 'Auth', type: :request do
+  let(:user) { build(:user) }
+  let(:sign_up_params) do
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/auth/sign_in_spec.rb]: sign_in_params

File: spec/request/api/v1/auth/sign_in_spec.rb [hunk 2]
```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+require 'faker'
+
+describe 'Auth', type: :request do
+  let(:user) { build(:user) }
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/auth/sign_in_spec.rb]: You should create the user to later test the login:
`let!(:user) { create(:user, email: 'test@example.com', password: 'password') }`

File: spec/request/api/v1/auth/sign_in_spec.rb [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  let(:params) do
+    {
+      email: user.email,
+      password: user.password
+    }
+  end
+
+  describe 'POST api/v1/auth/sign_in' do
+    before do
+      post api_v1_user_registration_path sign_up_params
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/auth/sign_in_spec.rb]: Be careful, here you're testing the sign_up endpoint.

File: spec/helpers.rb [hunk 4]
```diff
@@ -4,4 +4,21 @@ def json_value(data, *properties)
     properties.each { |property| value = value[property] }
     value
   end
+
+  def auth_header_from_response_headers(headers)
```
Comment (JuanfraM) [inline spec/helpers.rb]: As a suggestion, I think `auth_user_header` it's shorter and clearer

File: spec/helpers.rb [hunk 5]
```diff
@@ -4,4 +4,21 @@ def json_value(data, *properties)
...
+    uid = json_value(headers, 'uid')
+
+    auth_header = {
+      'access-token': access_token,
+      'client': client,
+      'uid': uid,
+      'expiry': expiry,
+      'token_type': token_type
+    }
+    auth_header
```
Comment (JuanfraM) [inline spec/helpers.rb]: isn't this redundant? before this sentence, you already have `auth_header`

File: spec/request/api/v1/auth/sign_in_spec.rb [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+      end
+
+      it 'is expected a successful response to other queries' do
+        auth_header = auth_header_from_response_headers response.headers
+        get api_v1_auth_validate_token_path, headers: auth_header
+        expect(response).to be_successful
+      end
+    end
+
+    context 'when the credentials are wrongs' do
```
Comment (JuanfraM) [inline spec/request/api/v1/auth/sign_in_spec.rb]: Tests are fine, but WDYT about this approach for contexts:
```
context 'when not valid' do
  context 'when credentials are missing' do
  end
  context 'when credentials are wrong' do
  end
end
```

**Review by gabrielbursztein2 — APPROVED** [review]
👏

**Review by JuanfraM — APPROVED** [review]

**Pull request merged**